overlord/hookstate/hookmgr.go

```diff
@@ -250,8 +253,30 @@ func MockRunHook(hookInvoke func(c *Context, tomb *tomb.Tomb) ([]byte, error)) (
 	}
 }
 
+var osReadlink = os.Readlink
+
+// snapCmd returns the "snap" command to run. If snapd is re-execed
+// it will be the snap command from the core snap, otherwise it will
+// be the system "snap" command (c.f. LP: #1668738).
+func snapCmd() string {
+	// sensible default, assume PATH is correct
+	snapCmd := "snap"
+
+	exe, err := osReadlink("/proc/self/exe")
+	if err != nil {
+		return snapCmd
```

Comment:
Don't we want to log something here? This may be denied by an LSM and we'd not know with a silent error like that.